test/common/Client.js

```diff
@@ -1360,4 +1360,78 @@ describe('The Client class', function() {
 
   });
 
+  describe('The destroyMessage method', function() {
+
+    it('should throw an Error if the id is undefined', function() {
+      expect(function() {
+        defaultClient().destroyMessage();
+      }).to.throw(Error);
+    });
+
+    it('should call setMessages with the expected id', function(done) {
+      var client = defaultClient();
+
+      client.transport.post = function(url, headers, body) {
+        expect(body).to.deep.equal([['setMessages', {
+          destroy: ['the id']
+        }, '#0']]);
+        done();
+
+        return q.reject();
+      };
+
+      client.destroyMessage('the id');
+    });
+
+    it('should resolve the promise with nothing is the message was destroyed', function(done) {
```

Comment:
should resolve the promise with nothing is the message was destroyed!!!! do you mean: if the message was ....